nixos/doc/manual/configuration/fhs.chapter.md

```diff
@@ -0,0 +1,199 @@
+# Binaries from other distributions {#sec-fhs}
+
+Software packaged in Nixpkgs is built to be platform-agnostic in the sense that
+it doesn't expect any dependencies to be implicitly provided by the host
+environment. If a Nix-built program needs to load dynamic shared libraries, the
+concrete revisions of these libraries are always deployed with the program in
+predictable locations. This way, a program built by Nix can be run on any other
+machine with the same architecture, regardless of the environment. For example,
+such a program doesn't need global paths like `/bin` or `/usr/lib` to exist,
+and it always loads the same versions of its dependencies.
+This determinism has known limits, cf. [](#sec-fhs-drivers).
+
+Programs distributed in the binary form outside Nixpkgs will often expect to
+find their dependencies in some of the several common global locations.
+Specifically they would expect to be run in the so-called "merged FHS
+environment" (for [Filesystem Hierarchy
+Standard](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard)). This
+concerns software downloaded using `pip` or `conda install`, toolchains fetched
+by build systems such as `gradle`, shell scripts with hard-coded paths, or
+anything downloaded from the web.
+
+In Nixpkgs the "FHS" hierarchy is splayed, rather than merged: every revision
+of a program is installed into its own prefix. NixOS does not materialize this
+hierarchy into any global paths, meaning that the user wishing to run a program
+expecting the "FHS" environment needs to explicitly point the program at its
+dependencies. NixOS provides several tools that automate or otherwise
+accommodate this task.
+
+## Dynamic linkage issues {#sec-fhs-dynamic-linkage}
+
+Dynamically linked native programs built for "FHS" environments expect to find
+the so-called "dynamic linker" in a fixed location, e.g.
+`/lib64/ld-linux-x86-64.so.2` depending on the platform. They might also expect
+a number of "system" libraries to be provided by the environment. In "FHS"
+distributions these are deployed in `/usr/lib`, and paths to them are
+discovered by the dynamic linker through the `/etc/ld.so.conf` file. These
+files are not available in NixOS by default.
+
+NixOS provides the `programs.nix-ld.enable` option which deploys a proxy where
```

Comment:
You immediately describe the workaround, without saying when to use this. I'd start this paragraph more like "If you want to run non-nix-built binaries without having to patch the interpreter field, you can set …, which does …"

And I'd probably also mention if it's okay to patch binaries, you're usually better off of using patchelf, or autopatchelfhook - and link to that manual section.

Reply:
I think there is a difference between just running and packaging those binaries. Running patchelf outside of a build might render the package useless after the next gc and is therefore nothing I'd recommend. There could be an article that focuses on packaging binaries like https://nixos.wiki/wiki/Packaging/Binaries but this feels out of scope of this PR.